subsys/mgmt/mcumgr/lib/mgmt/src/mgmt.c

```diff
@@ -200,15 +200,15 @@ mgmt_ctxt_init(struct mgmt_ctxt *ctxt, struct mgmt_streamer *streamer)
 void
 mgmt_ntoh_hdr(struct mgmt_hdr *hdr)
 {
-	hdr->nh_len = ntohs(hdr->nh_len);
-	hdr->nh_group = ntohs(hdr->nh_group);
+	hdr->nh_len = sys_cpu_to_be16(hdr->nh_len);
+	hdr->nh_group = sys_cpu_to_be16(hdr->nh_group);
 }
 
 void
 mgmt_hton_hdr(struct mgmt_hdr *hdr)
 {
-	hdr->nh_len = htons(hdr->nh_len);
-	hdr->nh_group = htons(hdr->nh_group);
+	hdr->nh_len = sys_be16_to_cpu(hdr->nh_len);
+	hdr->nh_group = sys_be16_to_cpu(hdr->nh_group);
```

Comment:
For int16 they both do the same thing but isn't this technically the wrong way around where mgmt_ntoh_hdr() should be BE16 to CPU and mgmt_hton_hdr() should be CPU to BE16?

Reply:
My bad.